packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -195,7 +196,7 @@ async function processRemoteNode({
 
   // Create the temp and permanent file names for the url.
   const digest = createHash(url)
-  const ext = path.parse(url).ext
+  const ext = path.parse(Url.parse(url).pathname).ext
```

Comment:
This is looking good to me! The basic idea is that it's going to make parsing the extension a little more robust, which is great.

How about some tests validating the approach? Also just to sanity check this, I [created a Runkit](https://runkit.com/dschau/5bad2e64a12ba30012a6adfa) to check this out!

Reply:
let's move `path.parse(Url.parse(url).pathname).ext` to separate function, that we can export and unit test against couple of urls including `https://scontent.xx.fbcdn.net/v/t51.2885-15/42078503_294439751160571_1602896118583132160_n.jpg?_nc_cat=101&oh=e30490a47409051c45dc3daacf616bc0&oe=5C5EA8EB`

Reply:
@oorestisime this is actually what I was going to suggest. If we move to a separate file/function, we can add some unit tests sorta like what I've done in the Runkit!

Reply:
yes sounds perfect. added this with latest commit. thanks